### PR TITLE
"""Reworks""" The Ancient Danger Ruin

### PR DESCRIPTION
**Pull request by goober3**
Revamped the Ancient Danger into an Ancient Complex on an island, also features a special subtype of space vine which can be destroyed in ONE chop instead of TWO chops.

The loots: -x4 rechargers
-x3 black & red spacesuits
-x2 laser rifles
-x2 laser accelerator cannons (with firing pins this time) -x1 binoculars
-x6 jackboots + x6 (7) combat gloves
-x1 syndie toolbox
-x1 exosuit fabricator
-x2 bluespace cells
-a sense of accomplishment

## BRING ME PICTURES OF THE RUIN

greebledvines
with the vines

greeblednovines
with no vines

greebled
in-game screenshot

greebledareas
with vines and maybe some areas.

ancientdangerbefore
and a picture of the original ruin for those that might be curious

does this look anything like the original ruin? absolutely not
is this still a rimworld reference? maybe.
is this marginally better than the original ruin? I sure do hope so

- image
 I affirm that I have tested all of my proposed changes and that any issues found during tested have been addressed.

## Why It's Good For The Game

Ancient Complex? It's actually quite simple.

## Changelog

:cl:
add: A new ancient complex ruin for beach planets
add: A weaker subtype for space vines
del: beach_ancient_ruin has been wiped off the face of ALL beach planets
/:cl:

### CODE REVIEW

**Comment by Zevotech** [comment]
> Ancient Complex? It's actually quite simple.

SPEEDMERGE

File: code/modules/events/spacevine.dm [hunk 1]
```diff
@@ -557,3 +557,7 @@
 	name = "dense space vines"
 	opacity = TRUE
 	icon_state = "Hvy1"
+
+/obj/structure/spacevine/weak
+    name = "weak space vines"
+    max_integrity = 25
```
Comment (Erikafox) [inline code/modules/events/spacevine.dm]: could you add a description to this subtype? Not required, but it's far fluffier than just calling the thing weak vine.

**Comment by Erikafox** [comment]
Excuse me I forgot that this was a ruin already. Just go through and make sure all the tags are still good.

File: code/game/area/areas/ruins/beachplanet.dm [hunk 2]
```diff
@@ -1 +1,16 @@
...
+
+/area/ruin/beach/complex/wall
+	name = "Cybersun Complex Exterior Wall"
+	icon_state = "red"
+	ambientsounds = MAINTENANCE
+
+/area/ruin/beach/complex/shuttle
+	name = "Syndicate Shuttle"
+	icon_state = "green"
+	ambientsounds = MAINTENANCE
```
Comment (Erikafox) [inline code/game/area/areas/ruins/beachplanet.dm]: you don't need to redefine the ambient sounds for each area, they'll inherit from /beach/complex

File: _maps/RandomRuins/BeachRuins/beach_ancient_ruin.dmm [hunk 3]
```diff
@@ -1,343 +1,8597 @@
...
+/turf/open/water/beach,
+/area/ruin/unpowered)
+"lR" = (
+/obj/machinery/light/small,
+/obj/structure/spacevine/weak,
+/turf/open/floor/concrete/slab_1,
+/area/ruin/beach/complex/wall)
+"lS" = (
+/turf/closed/wall/concrete{
+	desc = "As you draw near the ancient wall, a sense of foreboding overcomes you. You aren't sure why, but you feel this dusty structure may contain great dangers."
```
Comment (Erikafox) [inline _maps/RandomRuins/BeachRuins/beach_ancient_ruin.dmm:697]: you *could* make this an area based mood modifier.

**Comment by goober3** [comment]
ooh right, thank you for the heads up. catalogue's up to date now.

**Review by Erikafox — COMMENTED** [review]
The ocean area surrounding the island needs a bit more flavor, spawn in a beachworld and see what normally spawns there, and use that knowledge to decorate. 
StrongDMM_lN1OKDMQS7
These holofield gens won't work if someone were to get the shuttle working.
You could consider wiring all the exterior windows, so that if the ruin was powered, they'd shock anyone trying to break and enter.
If you're going to fluff the area as Cybersun Complex, I'd try to add some fluff items and decor to express that factionality. 
Also consider using the mapping helpers (obj/effect/mapping_helpers), specifically /abandoned and /locked on some of the airlocks. It'll make the ruin a bit more... Ruiny in my opinion. This room in particular could use the locked helper to make it a smidge harder to rush the goodies.
image
Overall not too much to say. It's a good rework.

File: _maps/RandomRuins/BeachRuins/beach_ancient_ruin.dmm (hunk 3 above)
Comment (goober3) [reply _maps/RandomRuins/BeachRuins/beach_ancient_ruin.dmm:697]: tried my best, moodlet message always broke when I changed the moodlet color from 'nicegreen' to 'warning'

Comment (Erikafox) [reply _maps/RandomRuins/BeachRuins/beach_ancient_ruin.dmm:697]: try 
mood_message = "<span class='boldwarning''>This stucture fills you with a sense of foreboding. You don't know how, but it seems dangerous.</span>\n"

**Comment by Latentish** [comment]
Aside from plating/beach/sand having a problem loading, this is good for merging.

**Pull request merged**